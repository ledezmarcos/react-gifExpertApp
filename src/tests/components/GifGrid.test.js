import React from "react";
import { shallow } from 'enzyme';
import { GifGrid } from "../../components/GifGrid";
import {useFetchGifs} from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente GifGrid',()=>{

    const category = 'Paraguay'
    const mockUseFetchGifs = useFetchGifs;
    
    test('Debe mostrarse correctamente ', () => {

        mockUseFetchGifs.mockImplementation( category => {
            return { data: [], loading: true }
          });
        const wrapper = shallow( <GifGrid category={category}/> );
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes con useFetchGifs', () => {
        const wrapper = shallow( <GifGrid category={category}/> );
        expect(wrapper).toMatchSnapshot();
        
        
    })
    
    

});
