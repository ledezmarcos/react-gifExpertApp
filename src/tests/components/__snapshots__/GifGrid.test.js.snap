// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Debe mostrarse correctamente  1`] = `
<Fragment>
  <h3>
    Paraguay
  </h3>
  <p>
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
